client/app/tasks/hooks/useTaskDetailsModal.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react'
+import { useRouter, useSearchParams } from 'next/navigation'
+import { TaskTypeWithUser } from '@/_types'
+import { CustomEventType } from '../types'
+
+export const useTaskDetailsModal = (tasks: TaskTypeWithUser[] | undefined) => {
+  const router = useRouter()
+  const searchParams = useSearchParams()
+  const taskId = searchParams.get('taskId')
```

Comment:
Seems like you are not using this taskId

Reply:
I'm using taskId in case you're accessing from the main TODO page. If you click a task from the TODO list, you pass a query like `?taskId=1`. In this case, you want to show immediately the modal

Reply:
Oh I confirmed it works now!